releasing/plugin-commands-deploy/test/deploy.test.ts

```diff
@@ -213,3 +213,53 @@ test('deploy with dedupePeerDependents=true ignores the value of dedupePeerDepen
   project.has('is-positive')
   expect(fs.existsSync('sub-dir/deploy')).toBe(false)
 })
+
+// Regression test for https://github.com/pnpm/pnpm/issues/8297 (pnpm deploy doesn't replace catalog: protocol)
+test('deploy works when workspace packages use catalog protocol', async () => {
```

Comment:
Created this test by copying the first test here and modifying it to use catalogs.

https://github.com/pnpm/pnpm/blob/a2e9bdcc028c729dc22fc9518650a34747fbeb8f/releasing/plugin-commands-deploy/test/deploy.test.ts#L10